### PR TITLE
Adds baikal-p7 5.6

### PR DESCRIPTION
**Pull request by mjvankampen**
Specify library name and version:  **baikal-p7/5.6**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/baical-p7/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+import os
+import shutil
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+
+required_conan_version = ">=1.28.0"
+
+class FlatbuffersConan(ConanFile):
+    name = "baical-p7"
+    license = "LGPL-3.0"
```
Comment (uilianries) [inline recipes/baical-p7/all/conanfile.py:11]:
```suggestion
    license = "LGPL-3.0-or-later"
```

File: recipes/baical-p7/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
+import glob
```
Comment (uilianries) [inline recipes/baical-p7/all/conanfile.py:1]:
```suggestion
```

no usage

File: recipes/baical-p7/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+import os
+import shutil
```
Comment (uilianries) [inline recipes/baical-p7/all/conanfile.py:3]:
```suggestion
```

no usage

File: recipes/baical-p7/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+
+required_conan_version = ">=1.28.0"
```
Comment (uilianries) [inline recipes/baical-p7/all/conanfile.py:7]:
```suggestion
required_conan_version = ">=1.29.0"
```

remove_files_by_mask is from 1.29.0

**Review by uilianries — CHANGES_REQUESTED** [review]

**Review by uilianries — APPROVED** [review]

File: recipes/baical-p7/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,75 @@
+import os
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+
+required_conan_version = ">=1.29.0"
```
Comment (prince-chrismc) [inline recipes/baical-p7/all/conanfile.py:5]:
```suggestion
required_conan_version = ">=1.29.1"
```

File: recipes/baical-p7/all/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+        if self.options.shared:
+            self.cpp_info.components["p7"].name = "p7-shared"
+            self.cpp_info.components["p7"].libs = ["p7-shared"]
+        else:
+            self.cpp_info.components["p7"].name = "p7"
+            self.cpp_info.components["p7"].libs = ["p7"]
+
+        if self.settings.os == "Linux":
+            self.cpp_info.components["p7"].system_libs .append("rt")
+            self.cpp_info.components["p7"].system_libs .append("pthread")
```
Comment (prince-chrismc) [inline recipes/baical-p7/all/conanfile.py:73]:
```suggestion
            self.cpp_info.components["p7"].system_libs .extends(["rt", "pthread"])
```

File: recipes/baical-p7/all/test_package/CMakeLists.txt [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+
+add_executable(${PROJECT_NAME} test_package.cpp)
+
+if(P7_SHARED)
+  target_link_libraries(${PROJECT_NAME} p7::p7-shared)
+else()
+  target_link_libraries(${PROJECT_NAME} p7::p7)
+endif()
+
+set_property(TARGET ${PROJECT_NAME} PROPERTY CXX_STANDARD 11)
```
Comment (prince-chrismc) [inline recipes/baical-p7/all/test_package/CMakeLists.txt:17]: correct me if i am wrong but I do not see any C++11 features in the test package

```suggestion
endif()
```

File: recipes/baical-p7/config.yml [hunk 8]
```diff
@@ -0,0 +1,4 @@
+---
```
Comment (prince-chrismc) [inline recipes/baical-p7/config.yml:1]:
```suggestion
```

**Review by prince-chrismc — CHANGES_REQUESTED** [review]
The only issue if the minimum but if you can please apply the other editorials that would be great!

File: recipes/baical-p7/all/test_package/CMakeLists.txt (hunk 7 above)
Comment (mjvankampen) [reply recipes/baical-p7/all/test_package/CMakeLists.txt]: Done! Somehow I couldn't commit your suggestions.

**Review by uilianries — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]
nice library name, I've been on the Baikal several times (it's a lake in Russia).

**Pull request merged**